src/sidebar/services/test/groups-test.js

```diff
@@ -182,6 +205,40 @@ describe('groups', function() {
         assert.calledWith(fakeApi.groups.list, sinon.match({ authority: 'publisher.org' }));
       });
     });
+
+    truthTable(3).forEach(([ loggedIn, pageHasAssociatedGroups, directLinkToPublicAnnotation ]) => {
```

Comment:
I'm generating the truth table programatically to ensure that no cases are missed.